### PR TITLE
Fix: Variables with '!' cant have default value

### PR DESCRIPTION
**Pull request by ashwin95r**
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/260)

### CODE REVIEW

File: gql/parser_test.go [hunk 1]
```diff
@@ -837,6 +837,17 @@ func TestParseVariablesError7(t *testing.T) {
...
 }
 
+func TestParseVariablesiError8(t *testing.T) {
+	query := `{
+		"query": "query testQuery($a: int = 3  , $b: int! =  4 ,  $c : int = 3){root(_uid_: 0x0a) {name(first: $b, after: $a, offset: $c){english}}}", 
+		"variables": {"$b": "5" } 
+	}`
+	_, _, err := Parse(query)
+	if err == nil {
+		t.Error("Vaiabes type ending with ! cant have default value")
```
Comment (pawanrawal) [inline gql/parser_test.go]: Variables. Also might be better to check for exact error?

**Comment by ashwin95r** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[gql/parser_test.go, line 847 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/260#-KT90Zqr-gEvTl8iXrFB-r1-847:-KT94q9l9de48WJsxNCq:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/8fcadec01ce8c95c47f34866b141fa62c5c5fe65/gql/parser_test.go#L847)):*
<details><summary><i>Previously, pawanrawal (Pawan Rawal) wrote…</i></summary>
> Variables. Also might be better to check for exact error?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/dgraph/260)*

**Comment by pawanrawal** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/dgraph/260#-:-KT95G2dqDyoLwvZdoay:bnfp4nl)*

**Comment by ashwin95r** [comment]
Merged to master with commit 2b6fddae65fe633ba75bbd8198333c2f5d321944.

**Pull request closed without merging**